status-im/status-react
# [WIP] Increase password check to 8 characters and add some basic weakness checks.

Fixes #13137

### Summary

Password limit should be increased to 8 characters and we should have some
checks to make sure users are not entering weak passwords.

In this PR, I've updated the minimum password length to 8 characters and added 2 weakness checks:
 - Make sure the password characters are not all repeated i.e. "aaaaaaaa" and "11111111" are not allowed
 - Make sure passwords are not sequential characters i.e. "abcdefgh" and "12345678" are not allowed

I've also created a password utility so that the onboarding screen and password reset screens can share
the same password validation code.

### Testing notes
Tested on android by inputting passwords to test failing conditions.
- entered 'aaaaaaaa'
- entered '12345678'
- entered other strings of characters

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
- Onboarding password creation
- Password reset

##### Functional
- new account
- user profile updates

### Steps to test
- Clear status storage to simulate being a new user
- Go through onboarding process and create a password, verify password meets minimum requirements mentioned in summary
- Change password and verify password meets minimum requirements mentioned in summary.



status: Ready
